te/TE_WORK.py

```diff
@@ -123,36 +127,57 @@ def tech_support(my_ip, remote_ip, remote_user, remote_pwd, remote_path, type_of
                 files_to_send.append((file_interested, destination))
                 folders_to_make.append(destination)
 
-        if(bin_src_flag):
-            if(type_of_logs == "all" or type_of_logs == "process" or type_of_logs=='core'):
-                file_interested_bin='/opt/te/bin/'
-                file_interested_src='/opt/te/src/'
-                file_interested_makefile='opt/te/Makefile'
-                destination=os.path.join(remote_path,'bin_src_file_dir/')
-                files_to_send.append((file_interested_bin, destination))
-                files_to_send.append((file_interested_src, destination))
-                files_to_send.append((file_interested_makefile, destination))
-                folders_to_make.append(destination)
-                
+        if(type_of_logs == "all" or type_of_logs == "process" or type_of_logs=='core'):
+            file_interested_bin='/opt/te/bin/'
+            file_interested_src='/opt/te/src/'
+            file_interested_makefile='opt/te/Makefile'
+            destination=os.path.join(folders_to_make[0],'bin_src_file_dir/')
+            files_to_send.append((file_interested_bin, destination))
+            files_to_send.append((file_interested_src, destination))
+            files_to_send.append((file_interested_makefile, destination))
+            folders_to_make.append(destination)
+
         if(bool(folders_to_make)):
-            str_folder_to_make = " ".join(folders_to_make[1:])
-            cmd = "sshpass -p %s ssh -o 'StrictHostKeyChecking no' -t %s@%s 'rm -rf %s; mkdir -p %s'" \
-                %(remote_pwd, remote_user, remote_ip, folders_to_make[0], str_folder_to_make)
+            str_folder_to_make = " ".join(folders_to_make)
+            cmd = "rm -rf %s; mkdir -p %s" %(folders_to_make[0], str_folder_to_make)
             (out, err) = __exec_cmd(cmd)
             lgr.info("Executing cmd=%s, out=%s, err=%s" %(cmd, out, err))
 
         for (src, dest) in files_to_send:
-            cmd = "sshpass -p %s scp -o 'StrictHostKeyChecking no' -r %s %s@%s:%s" \
-                %(remote_pwd, src, remote_user, remote_ip, dest)
+            cmd = "cp -r %s %s" %(src, dest)
             (out, err) = __exec_cmd(cmd)
-            lgr.info("Executing cmd=%s, out=%s, err=%s" %(cmd, out, err))    
+            lgr.info("Executing cmd=%s, out=%s, err=%s" %(cmd, out, err))
+
+        cmd = "tar -zcvf {} {}/*; rm -rf {}".format(tar_file_with_path, folders_to_make[0], folders_to_make[0])
+        (out, err) = __exec_cmd(cmd)
+        lgr.info("Executing cmd=%s, out=%s, err=%s" %(cmd, out, err))
+
+        try:
+            ssh = paramiko.SSHClient()
+            ssh.set_missing_host_key_policy(paramiko.AutoAddPolicy())
+            if remote_pwd:
+                ssh.connect(remote_ip, username=remote_user, password=remote_pwd)
+            else:
+                ssh.connect(remote_ip, username=remote_user)
+            scp = SCPClient(ssh.get_transport())
```

Comment:
Should we have a check if its not ssh enabled systems ( no password, no certs) then we don't even try the SCP logic

Reply:
If the system is not enabled for ssh, ssh.connect() would raise an exception which is caught in https://github.com/vmware/te-ns/pull/11/files/dd3075eadee0b381e4191045e88146e5446bca2e#diff-3b20c9cd5201723be91202ad245dbb93820190debf2ed38839a7a636c2d7b38dR176-R180